.golangci.yml

```diff
@@ -60,14 +63,24 @@ linters-settings:
       - alias: vmopv1a2sysprep
         pkg: github.com/vmware-tanzu/vm-operator/api/v1alpha2/sysprep
 
-      - alias: vmopv1
+      - alias: vmopv1a3
         pkg: github.com/vmware-tanzu/vm-operator/api/v1alpha3
-      - alias: vmopv1cloudinit
+      - alias: vmopv1a3cloudinit
         pkg: github.com/vmware-tanzu/vm-operator/api/v1alpha3/cloudinit
-      - alias: vmopv1common
+      - alias: vmopv1a3common
         pkg: github.com/vmware-tanzu/vm-operator/api/v1alpha3/common
-      - alias: vmopv1sysprep
+      - alias: vmopv1a3sysprep
         pkg: github.com/vmware-tanzu/vm-operator/api/v1alpha3/sysprep
```

Comment:
Seems like these aliases are not enforced as we have the following in the `api/v1alpha3/virtualmachine_bootstrap_types.go` file:
```
import (
	vmopv1cloudinit "github.com/vmware-tanzu/vm-operator/api/v1alpha3/cloudinit"
	vmopv1common "github.com/vmware-tanzu/vm-operator/api/v1alpha3/common"
	vmopv1sysprep "github.com/vmware-tanzu/vm-operator/api/v1alpha3/sysprep"
)
```

Reply:
I've noticed in the past that aliases are not always enforced everywhere, and I've never been able to figure out why. My guess is maybe it's because of the API package being a sub-module?

Reply:
And I just confirmed it's because it does not cross Go module boundaries. Unfortunately fixing that now would bring in a ton of new issues, so I will follow up on this in a separate commit.

Reply:
FWIW, there is no native support for loading partial configs or overwriting configs. The best I found is https://github.com/golangci/golangci-lint/discussions/2001. This matters because we may want rules that are specific to the `api` sub-module and not for the rest of the repo.